src/pools/abstract-pool.ts

```diff
@@ -222,6 +225,27 @@ export abstract class AbstractPool<
     )
   }
 
+  /** @inheritdoc */
+  public isPoolBusy (): boolean {
+    if (this.dynamic && this.workers.length === this.max) {
+      return true
+    } else if (this.findFreeTasksMapEntry() === [null, null]) {
+      return true
+    }
+    return false
+  }
+
+  /** @inheritdoc */
+  public findFreeTasksMapEntry (): [Worker, number] | [null, null] {
```

Comment:
I would prefer to return something like `undefined`, `null` or `false`  
So e.g.
```suggestion
  public findFreeTasksMapEntry (): [Worker, number] | false {
```